src/hooks/bookmarks/use-bookmarks-by-tutorial-ids.ts

```diff
@@ -0,0 +1,42 @@
+import { useQuery, UseQueryResult } from '@tanstack/react-query'
+import { ApiBookmark, ApiTutorial } from 'lib/learn-client/api/api-types'
+import { getBookmarksByTutorialIds } from 'lib/learn-client/api/bookmark'
+import useAuthentication from 'hooks/use-authentication'
+
+interface UseBookmarksByTutorialIdsOptions {
+	tutorialIds: ApiTutorial['id'][]
+}
+
+interface UseBookmarksByTutorialIdsResult extends Omit<UseQueryResult, 'data'> {
+	bookmarks: undefined | ApiBookmark[]
+}
+
+/**
+ * Handles fetching and storing bookmarks with React Query for the given
+ * `tutorialIds`.
+ */
+const useBookmarksByTutorialIds = ({
```

Comment:
I was considering that the page level view would fetch the specific set of tutorial bookmark data with this function. And then pass that down to the cards, instead of each card calling the hook. 🤔 Not that we have to do that but it was just my thought. 

It does seem a bit odd to me having every tutorial card calling the 'getAllBookmarks' hook. But maybe with the caching that's okay?....

Reply:
The reason I didn't go with that idea right away is because it requires invoking this hook on every view we have TutorialCards. So far, that means product landing pages, tutorial landing pages, collection views, and install landing pages. The cache is being leveraged correctly here too! Even across several pages, all bookmarks are only being fetched one time. I think your idea is a good optimization to try out in a separate PR since it'll require refactors in a quite a few places and this PR is already doing a lot of refactoring.